yare-model-validators/src/main/java/com/sabre/oss/yare/model/validator/DefaultRuleValidator.java

```diff
@@ -48,6 +54,8 @@ private static Validator createRuleValidator(boolean fastFail) {
                 ImmutableList.of(
                         new AttributeValidator(fastFail),
                         new FactValidator(fastFail),
+                        new ReservedValidator(fastFail, reservedNames),
+                        new DuplicateValidator(fastFail),
```

Comment:
We tend to add whole new Validator for some more complex checks e.g. Attributes. I think you should move those validations to ReferenceValidator class.